lib/orderbook/OrderBook.ts

```diff
@@ -154,31 +169,38 @@ class OrderBook extends EventEmitter {
     this.logger.debug(`order added: ${JSON.stringify(stampedOrder)}`);
   }
 
-  private updateOrderQuantity = (type: { [pairId: string]: Orders }, order: orders.StampedOrder, decreasedQuantity: number) => {
-    const object = this.getOrderMap(type, order);
-    object[order.id].quantity = object[order.id].quantity - decreasedQuantity;
-    if (object[order.id].quantity === 0) {
-      delete object[order.id];
+  private updateOrderQuantity = (type: Map<String, Orders>, order: orders.StampedOrder, decreasedQuantity: number) => {
+    const orderMap = this.getOrderMap(type, order);
+
+    orderMap[order.id].quantity = orderMap[order.id].quantity - decreasedQuantity;
+    if (orderMap[order.id].quantity === 0) {
+      delete orderMap[order.id];
     }
   }
 
-  private addOrder = (type: { [pairId: string]: Orders }, order: orders.StampedOrder) => {
+  private addOrder = (type: Map<String, Orders>, order: orders.StampedOrder) => {
+    if (order['localId']) {
+      this.localIdMap[order['localId']] = order.id;
+    }
+
     this.getOrderMap(type, order)[order.id] = order;
   }
 
-  private removeOrder = (type: { [pairId: string]: Orders }, orderId: string, pairId: string): boolean => {
+  private removeOrder = (type: Map<String, Orders>, orderId: string, pairId: string): boolean => {
```

Comment:
Any reason for not deleting the order id from `localIdMap`?

Reply:
No I completely forgot about that. Thanks!